modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/internal/servlet/taglib/BaseContainerTag.java

```diff
@@ -390,6 +411,12 @@ componentDescriptor, prepareProps(new HashMap<>()),
 		return EVAL_PAGE;
 	}
 
+	protected void processStartBodyTag() throws Exception {
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write("<div class=\"tag-body-content\">");
```

Comment:
couldn't we leverage the processCssClass() here to add `tag-body-content`?

Reply:
Nope, those are two different things:

- `processCssClass` adds classes to the tag main element
- This `tag-body-content` needs to be added in the body container, which is an element inside the main element.

An example with ButtonTag:

ButtonTag main element is a `button`, which is set in `doStartTag` with `setContainerElement("button");`. We define this element classes in `processCssClass`. Inside that node we're rendering stuff like icons or label if they're present. The the body is rendered, so the markup result would be something like

```
<button class="classesDefinedInProcessCssClass">
      <svg iconstuff />
      BODYMARKUP
</button>
```

What we need to get is just that BODYMARKUP, so we're adding a div with `tag-body-content` class to isolate it and make possible to query it.

```
<button class="classesDefinedInProcessCssClass">
      <svg iconstuff />
      <div class="tag-body-content">
            BODYMARKUP
      </div>
</button>
```

Reply:
I see, so we need what you implemented [here](https://github.com/liferay-frontend/liferay-portal/pull/3045/commits/8812a3102832215a262612dcb712544a231de5db) to deal with this.